lexicon/providers/aurora.py

```diff
@@ -22,17 +22,17 @@ def ProviderParser(subparser):
 
 class Provider(BaseProvider):
 
-    def __init__(self, options, engine_overrides=None):
-        super(Provider, self).__init__(options, engine_overrides)
+    def __init__(self, config):
+        super(Provider, self).__init__(config)
         self.domain_id = None
-        self.api_endpoint = self.engine_overrides.get('api_endpoint', 'https://api.auroradns.eu')
+        self.api_endpoint = 'https://api.auroradns.eu'
```

Comment:
hmm, is see this change in multiple providers. How do you override the api_endpoint for testing if you don't support it in the provider?

Reply:
In fact there is several things here.

Now basically, when you want to override an option, you implement `_test_parameters_overrides` in the relevant test class, and optionally `_test_fallback_fn` when you want to modify what default value will be given to any option that does not have a value (from environment variables or config file, in the case of integration tests.

Now, I saw a lot of code that would override parameters in a given provider, but with these parameters that are not formerly exposed options. For instance, a lot of `api_endpoint` parameters that cannot be set by command line, environment variables or configuration files. A logic is just hardcoded in the provider class to allow to modify the value during tests. At least I guesse e there is no clue that there were another value beside the default one during tests.

I do not like this, because you push code in the provider class that is here only to make the test be runnable. In theory, you should not modify your codebase to test it, you have mocks for that. And here `_test_parameters_overrides` and `_test_fallback_fn` are these kind of mocks.

I took a really opinionated approach. I decided to remove all codes from provider classes that were obviously here only for the tests. And to not provide any particular logic to modify provider parameters for tests only. Currently, any parameter that needs to be overriden during tests is required to be a real configuration parameter, so settable through command line, environment variable or configuration file.

If someone needs to relaunch integration tests against a real test API that does not provide a configurable `api_endpoint`, he will need to implement the logic on the provider parser, the provider class and the provider test.

I know this is a quite radical approach, but I think that being consistent about the code of the provider toward the code of the test will improve the code quality.

Reply:
Hey @adferrand 
I understand your reluctance to have test related code sprinkled in the providers. There was definitely come copy-paste during development, but i thought having it in the provider by default gave third party developers something to use as an example when they started their own providers. 

You're right though, its not really used consistently and is confusing and unnecessary for some providers.